server/app/services/special-account-service.js

```diff
@@ -116,12 +130,13 @@ async function updateSpecialAccountById(specialAccountId, updatedSpecialAccount,
         const updateCo = updatedSpecialAccount.connection;
 
         // If connection information is changed
+        // TODO implement username update
         if (updateCo) {
             const co = await currentSpecialAccount.getConnection();
             await co.update(
                 cleanObject({
-                    username: updateCo.username,
-                    password: await hash(updateCo.password)
+                    username: updateCo.username ? updateCo.username : co.username,
```

Comment:
No need for that I think ? the `cleanObject` function remove all field at `undefined`

Reply:
Also, we should see to prevent null values with Joi

Reply:
Yep, i gonna look for prevent nul value with JOI